node/libs/roles/src/attester/messages/batch.rs

```diff
@@ -160,46 +156,43 @@ impl BatchQC {
 
     /// Add a attester's signature.
     /// Signature is assumed to be already verified.
-    pub fn add(&mut self, msg: &Signed<Batch>, genesis: &Genesis) -> anyhow::Result<()> {
-        use BatchQCAddError as Error;
-
-        let committee = genesis
-            .attesters
-            .as_ref()
-            .context("no attester committee in genesis")?;
-
-        ensure!(self.message == msg.msg, Error::InconsistentMessages);
-        ensure!(!self.signatures.contains(&msg.key), Error::Exists);
-        ensure!(
-            committee.contains(&msg.key),
-            Error::SignerNotInCommittee {
-                signer: Box::new(msg.key.clone()),
-            }
+    pub fn add(
+        &mut self,
+        msg: &Signed<Batch>,
+        committee: &attester::Committee,
+    ) -> anyhow::Result<()> {
+        anyhow::ensure!(self.message == msg.msg, "inconsistent messages");
+        anyhow::ensure!(
+            !self.signatures.contains(&msg.key),
+            "signature already present"
         );
-
+        anyhow::ensure!(committee.contains(&msg.key), "not in committee");
```

Comment:
What's the reason to stop using `Error::InconsistentMessage` and the other errors, but not to remove the enum?

Reply:
ok, this method was fucked up - it was using and enum and wrapping it immediately into anyhow. Given that there is no use case for distinguishing the error programmatically, I'll remove the enum.